KD_Lib/Quantization/qat/qat.py

```diff
@@ -0,0 +1,126 @@
+import torch
+from KD_Lib.Quantization.common import Quantizer
+from copy import deepcopy
+
+
+class QAT_Quantizer(Quantizer):
+    """
+    Implementation of Quantization-Aware Training (QAT) for PyTorch models.
+
+    :param model (torch.nn.Module): (Quantizable) Model that needs to be quantized
+    """
+
+    def __init__(self, model):
+        super(QAT_Quantizer, self).__init__(model)
+
+    def quantize(
+        self,
+        train_loader,
+        test_loader,
+        model_optimizer,
+        criterion,
+        num_train_epochs=10,
+        num_train_batches=10,
+        param_freeze_epoch=3,
+        bn_freeze_epoch=2,
+        qconfig=torch.quantization.get_default_qat_qconfig("fbgemm"),
+        device=torch.device("cpu"),
```

Comment:
Shouldn't we be taking these parameters in the `__init__` method rather than taking it here

Reply:
Put this stuff here so that the code is similar to the other methods. I'll add the train, test loader and stuff in init